Tests/MongoSwiftTests/SDAMMonitoringTests.swift

```diff
@@ -52,32 +52,33 @@ final class SDAMTests: MongoSwiftTestCase {
 
             receivedEvents.append(event)
         }
+
         // do some basic operations
         let db = client.db(type(of: self).testDatabase)
         _ = try db.createCollection(self.getCollectionName())
         try db.drop()
 
         center.removeObserver(observer)
 
-        var error = bson_error_t()
-        guard let uri = mongoc_uri_new_with_error(MongoSwiftTestCase.connStr, &error) else {
-            XCTFail(toErrorString(error))
-            return
-        }
+        var opts = ClientOptions()
+        let connString = try ConnectionString(MongoSwiftTestCase.connStr, options: &opts)
 
-        guard let hostlist = mongoc_uri_get_hosts(uri) else {
-            XCTFail("Could not get hostlists for uri: \(uri)")
+        guard let hostlist = mongoc_uri_get_hosts(connString._uri) else {
+            XCTFail("Could not get hostlists for uri: \(MongoSwiftTestCase.connStr)")
             return
         }
 
         // check event count and that events are of the expected types
-        expect(receivedEvents.count).to(equal(5))
+        expect(receivedEvents.count).to(equal(7))
         expect(receivedEvents[0]).to(beAnInstanceOf(TopologyOpeningEvent.self))
         expect(receivedEvents[1]).to(beAnInstanceOf(TopologyDescriptionChangedEvent.self))
         expect(receivedEvents[2]).to(beAnInstanceOf(ServerOpeningEvent.self))
         expect(receivedEvents[3]).to(beAnInstanceOf(ServerDescriptionChangedEvent.self))
         expect(receivedEvents[4]).to(beAnInstanceOf(TopologyDescriptionChangedEvent.self))
 
+        // we are ignoring the last two events here. I think they are being generated by mongoc in error.
```

Comment:
asked in #mongoc and waiting to hear back. 
this test is based on the standalone test case in the SDAM yaml tests. 

the extra events are a ServerChangedEvent and a TopologyChangedEvent but in both of them the previous and new descriptions are identical.
I think it's likely that these events are generated in error but libmongoc doesn't notice because they only look at the first 5 events that the yaml test says should be there.

Reply:
update: this is a CDRIVER bug, [CDRIVER-3256](https://jira.mongodb.org/browse/CDRIVER-3256). they weren't running the SDAM monitoring tests in pooled mode so they missed this.

the spec says the events should not be emitted when the old and new descriptions are the same, so I have updated our SDAM monitoring code so that it compares the descriptions for the changed events, and does not publish the event if they are the same.

changed this test back to expecting 5 events.